[NC-675] add onchange behavior to text area native widget

Ahmad-Elsayed opened the pull request:
Feature is not released yet

Ahmad-Elsayed committed:
add onchange behavior to text area widget

pratikgoyal9 commented on content/en/docs/refguide/modeling/pages/input-widgets/text-area.md:
```diff
@@ -59,13 +59,37 @@ The on-change property specifies an action that will be executed when leaving th
+
+##### 2.5.2.1 When user leaves input field (Default)
+
+This option will work as in previous versions of Studio Pro. Text Area will apply changes when a value is not the same as previously saved value in the database and one of the following conditions is met:
```
Will this option work with previous versions of Studio Pro? No, right?

pratikgoyal9 replied:
Fixed, should be resolved!

pratikgoyal9 commented on content/en/docs/refguide/modeling/pages/input-widgets/text-area.md:
```diff
@@ -59,13 +59,37 @@ The on-change property specifies an action that will be executed when leaving th
+
+This option will work as in previous versions of Studio Pro. Text Area will apply changes when a value is not the same as previously saved value in the database and one of the following conditions is met:
+
+* On enter key pressed: This will trigger on change and on enter key press events
```
On press of the enter key, no `onChangeBehavior` would be triggered in a text area. Instead, the user should be navigated to the next line in the same text area. So, user does not leave the current text area in this context. 

According to me, the user can only leave the textarea by pressing the tab key(in case of the web) or by tapping any other part of the screen except in the current text area. So, this event should be removed.

@jordanch @cjfhodges Would you like to add your two cents?

jordanch replied:
Makes sense. Maybe missed after copying from [text box](https://github.com/mendix/docs/blob/4487/content/en/docs/refguide/modeling/pages/input-widgets/text-box.md).

pratikgoyal9 replied:
Fixed, should be resolved!

pratikgoyal9 commented on content/en/docs/refguide/modeling/pages/input-widgets/text-area.md:
```diff
@@ -59,13 +59,37 @@ The on-change property specifies an action that will be executed when leaving th
+This option will work as in previous versions of Studio Pro. Text Area will apply changes when a value is not the same as previously saved value in the database and one of the following conditions is met:
+
+* On enter key pressed: This will trigger on change and on enter key press events
+* Blurred: This will trigger on change and on leave events
```
I've no idea how this would work, where can I find and test the `Blurred` feature?

Ahmad-Elsayed replied:
when you touch screen outside the input and the keyboard, the input shouldn't be active, keyboard should be dismissed and then the `onChange` event should be called if `apply changes` property is set to `when user leaves input field`

pratikgoyal9 replied:
Ah okay! Good to know that it's called a `Blurred` condition.

pratikgoyal9 replied:
Fixed, should be resolved!

Ahmad-Elsayed committed:
update when user leaves input description

pratikgoyal9 commented on content/en/docs/refguide/modeling/pages/input-widgets/text-area.md:
```diff
@@ -59,13 +59,36 @@ The on-change property specifies an action that will be executed when leaving th
+
+##### 2.5.2.1 When user leaves input field (Default)
+
+Text Area will apply changes when a value is not the same as previously saved value in the database and one of the following condition is met:
```
Since we removed the other condition, it should be like the below:

`Text Area will apply changes when a value is not the same as previously saved value in the database and the following condition is met:`

pratikgoyal9 replied:
Fixed!

Ahmad-Elsayed committed:
update when user leaves input description

pratikgoyal9 reviewed (approved):
Good work @Ahmad-Elsayed 👍

ConnorLand committed:
Proofread

ConnorLand committed:
Make hyphens conform to UI text

On change etc is not hyphenated in product text, so I am removing the hyphens from documentation.

ConnorLand committed:
Proofread

ConnorLand committed:
Formatting

Ahmad-Elsayed committed:
update 'While user is entering data' description

ConnorLand committed:
Edit

ConnorLand committed:
Add break

ConnorLand committed:
Fix typo
